inst/@symfun/mtimes.m

```diff
@@ -58,7 +58,8 @@
 
 function h = mtimes(f, g)
   [vars, s1, s2] = helper_symfun_binops(f, g);
-  h = symfun(s1 * s2, vars);
+  res = s1 * s2;
+  h = symfun(res, symvar(res));
```

Comment:
> I can change it to take union of the variables from both the operands involved.  

Let's take this conversation to #738.  I don't see it being relevant to this PR.